src/gleam/list.gleam

```diff
@@ -790,7 +790,19 @@ pub fn find_map(
 pub fn all(in list: List(a), satisfying predicate: fn(a) -> Bool) -> Bool {
   case list {
     [] -> True
-    [x, ..rest] -> predicate(x) && all(rest, predicate)
+    list -> all_tail_recursive(list, predicate, True)
+  }
+}
+
+fn all_tail_recursive(
```

Comment:
Can we call these helpers `do_any` and `do_all`? That's the convention currently.